mlflow/utils/environment.py

```diff
@@ -442,11 +445,40 @@ def _is_mlflow_requirement(requirement_string):
             # Try again with the per-requirement options removed
             return Requirement(requirement_specifier).name.lower() == "mlflow"
         except InvalidRequirement:
-            return False
 
-        # TODO: Return True if `requirement_string` represents a project directory for MLflow
-        # (e.g. '/path/to/mlflow') or git repository URL (e.g. 'https://github.com/mlflow/mlflow').
-        return False
+            # Support defining branch dependencies for local builds or direct GitHub builds
+            # from source.
+            # Example: mlflow @ git+https://github.com/mlflow/mlflow@branch_2.0
+            repository_matches = ["/mlflow", "mlflow@git"]
+
+            return any(
+                match in requirement_string.replace(" ", "").lower() for match in repository_matches
+            )
+
+
+def _generate_mlflow_version_pinning(strict=False):
```

Comment:
Do we need `strict`? Do we set `strict` to `True` somewhere?

Reply:
No, I left it in here as a question for the PR review - should we do:
a) major pinning
b) minor pinning
c) strict micro pinning
Thoughts?
If a) -> I'll remove the arg and simplify the function.

Reply:
Awesome, just major pinning. We should use `mlflow<3` for all models logged in MLflow 2.0. Note that `mlflow~=2` is actually error prone due to quirks in Python / PyPI's dependency resolution procedure, so `mlflow<3` is preferred.

Reply:
awesome! I'll make that change :)